src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,441 @@
+                                for summary sheet and questions, references,
+                                and responses sheet.
+        excel_file (str):       Name of excel file to be created the summary
+                                and dataset sheets.
+    Returns:
+        None
+    """
+    summary_df = pd.DataFrame()
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    summary_df["scores"] = scores
+
+    question_indices = [f"Q{i + 1}" for i in range(len(summary_df))]
+    question_indices.append("Average")
+    question_indices.append("Total Score")
+    question_indices.append("Median")
+
+    average = summary_df.mean(axis=0, numeric_only=True)
+    total_score = summary_df.sum(axis=0, numeric_only=True)
+    median = summary_df.median(axis=0, numeric_only=True)
+
+    summary_df.loc[len(summary_df)] = average
+    summary_df.loc[len(summary_df)] = total_score
+    summary_df.loc[len(summary_df)] = median
+
+    summary_df.insert(0, "question index", question_indices)
+
+    response_df = result.dataset.to_pandas()
+    response_df["scores"] = scores
+
+    with pd.ExcelWriter(excel_file) as writer:
+        # df with contents similar to those in the .jsonl and .csv output files
+        response_df.to_excel(writer, sheet_name="dataset", index=False)
+        summary_df.to_excel(writer, sheet_name="Summary", index=False)
+
+
+def write_results(
+    result: EvaluationResult, file_formats: List[str], output_dir: str, model_name: str
+) -> List[str]:
+    """
+    Writes results files for DK-Bench for each file format provided.
+    Files have the name {output_dir}/responses-{model_name}-{timestamp}.{file_type}"
+
+    Each of the entries in a file has the following fields:
+    {model_name, scores, evaluation_run, user_input, response, reference}
+
+    Args:
+        result (Evaluation):      ragas EvaluationResult to parse for scores
+                                  for summary sheet and questions, references,
+                                  and responses sheet.
+        file_formats (List[str]): List of file formats of files to write results to.
+        output_dir (str):         Directory for results to be written out to
+        model_name (str):         Model name to be used in the results file name.
+    Returns:
+        List[str]:                A list of strings that are the file names of the files
+                                  with responses and scores for DK-Bench
+    """
+    validate_output_file_formats(file_formats)
+    response_df = result.dataset.to_pandas()
+    response_df["model_name"] = model_name
+
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    response_df["scores"] = scores
+
+    timestamp = datetime.now().isoformat()
+    response_df["evaluation_run"] = f"run-{timestamp}"
+
+    results_files = []
+    for fmt in file_formats:
+        if IOFileType.JSONL.value == fmt:
+            results_file = create_results_file_name(
```

Comment:
It seems like there's a logic being duplicated on either side of this function call to figure out which file extension is being saved to. Could we please just get rid of this function entirely and do the filename formatting up above? We can just supply the extension afterwards as `outfile_path = f"{results_file}.{extension}"`